### PR TITLE
Add Colyseus v0.13 monkey patch to fix deployment to a subroute

### PR DESCRIPTION
**Pull request by pjanik**
[[#184539868]](https://www.pivotaltracker.com/story/show/184539868)

This PR adds a Colyseus v0.13 monkey patch. Updated README should explain the problem.

I could create our fork of Colyseus, but instead, I've tried to use `patch-package` tool:
https://www.npmjs.com/package/patch-package

It works pretty neatly. You can modify the source directly in `node_modules` dir, then call `npx patch-package <problematic-dependency-name>` and it'll automatically create a patch code in `patches` dir. This patch is automatically applied in the npm `postinstall` script.

I've also updated Docker files to make sure `postinstall` script is applied (and patch source is available while NPM is installing packages).

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
 
 WORKDIR /CEASAR-server
-COPY package*.json ./
-
-RUN npm install
 
+# Copy complete source so node_module mokey-patch source is available too (applies in npm run postinstall).
 COPY . .
 
+RUN npm install
```
Comment (dougmartin) [inline Dockerfile:8]: Same comment as last PR, maybe use `npm ci`?

File: README.md [hunk 2]
```diff
@@ -6,20 +6,37 @@ A multiplayer game server for the [CEASAR](https://github.com/concord-consortium
...
 * If things are busted, try a nuke-and-repave: `./repave.sh`
 
+## Colyseus v0.13 monkey patch
+
+Make sure that `npm postinstall` script is executed (it should happen automatically, but you can also do it manually).
+It applies a monkey patch to Colyseus server v0.13 that lets us deploy server to a subroute:
+
+https://apps.concord.org/ceasar-server
+
+Otherwise, this app would need to be deplpyed to a top-level path (htts://ceasar-server.concord.org), which doesn't
```
Comment (dougmartin) [inline README.md:16]: deplpyed => deployed

File: README.md [hunk 3]
```diff
@@ -6,20 +6,37 @@ A multiplayer game server for the [CEASAR](https://github.com/concord-consortium
...
-You will need to obtain that URI from another developer or from
-Config Var of the same name in the Heroku [ceaser-server-staging app](https://dashboard.heroku.com/apps/ceaser-server-staging/settings).
+The only variable necessary to set is BASE_PATH. It can be set to / or any other path that you use in the front-end app.
 
 ### Running Locally:
 
 To start a live-reloading development server run:
 ```
-   npm run live
+  npm install
```
Comment (dougmartin) [inline README.md:38]: `npm ci`?

**Review by dougmartin — APPROVED** [review]
Looks good :+1: Just same suggestion as last PR for `npm ci` plus a spelling correction.  The patch system looks like an elegant solution - I'd think you would want to use `npm ci` now to ensure the patch is patching the same code.

**Comment by pjanik** [comment]
Thanks, I've just fixed the typo and replaced npm `install` with `ci` in the last commit.

**Pull request merged**